server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -0,0 +1,251 @@
+
+    private static final Logger logger = LogManager.getLogger(LocalHealthMonitor.class);
+
+    public static final Setting<TimeValue> INTERVAL_SETTING = Setting.timeSetting(
+        "health.reporting.local.monitor.interval",
+        TimeValue.timeValueSeconds(30),
+        TimeValue.timeValueSeconds(10),
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
+    private final ClusterService clusterService;
+    private final ThreadPool threadPool;
+    private final DiskCheck diskCheck;
+
+    private volatile TimeValue monitorInterval;
+    private volatile Scheduler.ScheduledCancellable scheduled;
+    private volatile IndividualNodeHealth lastReportedHealth = null;
+    private volatile boolean enabled;
+    private volatile boolean healthMetadataInitialized;
+
+    public LocalHealthMonitor(Settings settings, ClusterService clusterService, NodeService nodeService, ThreadPool threadPool) {
+        this.threadPool = threadPool;
+        this.monitorInterval = INTERVAL_SETTING.get(settings);
+        this.enabled = HealthNodeTaskExecutor.ENABLED_SETTING.get(settings);
+        this.clusterService = clusterService;
+        this.diskCheck = new DiskCheck(nodeService);
+        clusterService.addListener(this);
+        ClusterSettings clusterSettings = clusterService.getClusterSettings();
+        clusterSettings.addSettingsUpdateConsumer(INTERVAL_SETTING, this::setMonitorInterval);
+        clusterSettings.addSettingsUpdateConsumer(HealthNodeTaskExecutor.ENABLED_SETTING, this::setEnabled);
+    }
+
+    void setMonitorInterval(TimeValue monitorInterval) {
+        this.monitorInterval = monitorInterval;
+        if (scheduled != null && scheduled.cancel()) {
+            scheduleNextRunIfEnabled(new TimeValue(1));
+        }
+    }
+
+    void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        if (scheduled != null) {
+            scheduled.cancel();
+            if (enabled) {
+                scheduleNextRunIfEnabled(new TimeValue(1));
+            }
+        }
+    }
+
+    private void scheduleNextRunIfEnabled(TimeValue time) {
+        if (enabled && threadPool.scheduler().isShutdown() == false) {
+            scheduled = threadPool.schedule(this::monitorHealth, time, ThreadPool.Names.MANAGEMENT);
+        }
+    }
+
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        // Wait until every node in the cluster is upgraded to 8.4.0 or later
+        if (event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_4_0)) {
+            // Wait until the health metadata is available in the cluster state
+            if (healthMetadataInitialized == false) {
+                healthMetadataInitialized = HealthMetadata.getFromClusterState(event.state()) != null;
+                if (healthMetadataInitialized) {
+                    scheduleNextRunIfEnabled(TimeValue.timeValueMillis(1));
```

Comment:
I think this should schedule based the monitoring based on the normal interval

Reply:
hm, you mean that if it gets rescheduled it should have the initial delay? 

I have to admit that when I see something executed at an interval I do not expect to have an initial delay. I expect that it's going to be executed asap and then wait for the interval before the next execution. For example let's say that the interval is 10 minutes. I would think it's wrong to wait for 10 minutes before running it for the first time.

What do you think?